File: labs/OOP/lab1/code/src/core/FormulaDriver.cpp
#include "FormulaDriver.h"

FormulaDriver::FormulaDriver(const std::string& fName, const std::string& lName, int number, int pts)
    : firstName(fName), lastName(lName), racingNumber(number), points(pts) {}

std::string FormulaDriver::getFirstName() const {
    return firstName;
}

std::string FormulaDriver::getLastName() const {
    return lastName;
}

std::string FormulaDriver::getFullName() const {
    return firstName + " " + lastName;
}

int FormulaDriver::getRacingNumber() const {
    return racingNumber;
}

int FormulaDriver::getPoints() const {
    return points;
}

void FormulaDriver::addPoints(int pts) {
    points += pts;
}

File: labs/OOP/lab2/code/src/Core/FormulaDriver.cs
using System;

namespace RacingSystem.Core
{
  public class FormulaDriver
  {
    public int Id { get; set; }
    public int TeamId { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public int RacingNumber { get; set; }
    public int Points { get; set; }

    public FormulaDriver() : this(string.Empty, string.Empty, 0, 0) { }

    public FormulaDriver(string firstName, string lastName, int number, int points = 0)
    {
      FirstName = firstName;
      LastName = lastName;
      RacingNumber = number;
      Points = points;
    }

    public string FullName => $"{FirstName} {LastName}";
    public string DisplayString => $"{FullName} #{RacingNumber}";

    public void AddPoints(int pts) => Points += pts;
  }
}
